clusterloader2/pkg/tuningset/randomized_time_limited.go

```diff
@@ -0,0 +1,48 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package tuningset
+
+import (
+	"math/rand"
+	"time"
+
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/perf-tests/clusterloader2/api"
+)
+
+type randomizedTimeLimitedLoad struct {
+	params *api.RandomizedTimeLimitedLoad
+}
+
+func newRandomizedTimeLimitedLoad(params *api.RandomizedTimeLimitedLoad) TuningSet {
+	return &randomizedTimeLimitedLoad{
+		params: params,
+	}
+}
+
+func (r *randomizedTimeLimitedLoad) Execute(actions []func()) {
+	var wg wait.Group
+	for i := range actions {
+		index := i
+		wg.Start(func() {
+			// Sleeps for random duration in [0, TimeLimit].
+			time.Sleep(time.Duration(rand.Intn(int(r.params.TimeLimit))))
```

Comment:
This may not work - time.Duration is converted to nanoseconds, and int can be 4 bytes...
Should be sth like:

time.Sleep(time.Duration(rand.Int63n(r.params.TimeLimit.Nanoseconds()))

Reply:
Done.